templates/vue/src/components/TapestryMap.vue

```diff
@@ -0,0 +1,324 @@
+        :options="mapOptions"
+        :bounds="getBounds"
+        :max-bounds="worldBounds"
+        @update:center="updateCenter"
+        @update:zoom="updateZoom"
+      >
+        <l-tile-layer :url="url" :attribution="attribution" />
+        <l-marker
+          v-for="marker in markerlocations"
+          :key="marker.id"
+          :ref="'marker-' + marker.id"
+          :lat-lng="marker.pos"
+          :icon="
+            marker.status == 'draft'
+              ? draftIcon
+              : marker.accessible
+              ? icon
+              : inaccessibleIcon
+          "
+          @click="selectNode(marker.id)"
+        >
+          <l-popup>
+            <b-button
+              v-if="marker.accessible || canEdit"
+              variant="link"
+              @click="openNode(marker.id)"
+            >
+              <h6>{{ marker.title }}</h6>
+            </b-button>
+            <div v-else>
+              <h6>{{ marker.title }}</h6>
+            </div>
+          </l-popup>
+        </l-marker>
+      </l-map>
+    </div>
+  </div>
+</template>
+
+<script>
+import "leaflet/dist/leaflet.css"
+import mapMarker from "@/assets/map-marker.png"
+import invalidMarker from "@/assets/map-marker-inaccessible.png"
+import draftMarker from "@/assets/map-marker-draft.png"
+import { latLng, latLngBounds, icon } from "leaflet"
+import { LMap, LTileLayer, LMarker, LPopup } from "vue2-leaflet"
+import { mapState } from "vuex"
+import { names } from "@/config/routes"
+import Helpers from "@/utils/Helpers"
+import { isLoggedIn } from "@/utils/wp"
+
+export default {
+  name: "tapestry-map",
+  components: {
+    LMap,
+    LTileLayer,
+    LMarker,
+    LPopup,
+  },
+  props: {
+    isSidebarOpen: {
+      type: Boolean,
+      required: false,
+    },
+  },
+  data() {
+    return {
+      url: "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
+      attribution:
+        '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
+      mapOptions: {
+        zoomSnap: 0.1,
+        scrollWheelZoom: false,
+      },
+      icon: icon({
+        iconUrl: mapMarker,
+        iconSize: [32, 33],
+        iconAnchor: [16, 32],
+        popupAnchor: [0, -20],
+      }),
+      inaccessibleIcon: icon({
+        iconUrl: invalidMarker,
+        iconSize: [32, 33],
+        iconAnchor: [16, 32],
+        popupAnchor: [0, -20],
+      }),
+      draftIcon: icon({
+        iconUrl: draftMarker,
+        iconSize: [32, 33],
+        iconAnchor: [16, 32],
+        popupAnchor: [0, -20],
+      }),
+      worldBounds: latLngBounds([
+        [-90, -180],
+        [90, 180],
+      ]),
+    }
+  },
+  computed: {
+    ...mapState(["settings", "nodes", "rootId"]),
+    empty() {
+      return Object.keys(this.nodes).length === 0
+    },
+    selectedId() {
+      return this.$route.params.nodeId
+    },
+    isLoggedIn() {
+      return isLoggedIn
+    },
+    canEdit() {
+      return wpData.wpCanEditTapestry === "1"
+    },
```

Comment:
Can you change this to use the [wp module](https://github.com/wynnset/tapestry-wp/blob/master/templates/vue/src/services/wp.js) instead for consistency?